src/LedgerWallet.js

```diff
@@ -249,29 +234,35 @@ class LedgerWallet {
       // EIP155: v should be chain_id * 2 + {35, 36}
       const signedChainId = Math.floor((tx.v[0] - 35) / 2);
       if (signedChainId !== chainId) {
-        return Promise.reject(
-          new Error(
-            "Invalid signature received. Please update your Ledger Nano S."
-          )
+        throw new Error(
+          "Invalid signature received. Please update your Ledger Nano S."
         );
       }
 
       // Return the signed raw transaction
-      const rawTx = `0x${tx.serialize().toString("hex")}`;
-      return Promise.resolve(rawTx);
-    } catch (e) {
-      return Promise.reject(e);
+      return `0x${tx.serialize().toString("hex")}`;
     } finally {
       if (eth !== null) {
         // This is fishy but currently ledger library always returns empty
         // resolved promise when closing connection so there is no point in
         // doing anything with returned Promise.
         // noinspection JSIgnoredPromiseFromCall
```

Comment:
are these noinspection lines still valid? you have await on the result. And on the other hand I would not pollute our codebase with InteliJ products specific comments. You shall silent it somehow in you project config. We should stick to eslint for linting.

Reply:
You are totally right now its not required but anyway it should never get there.